packages/datetime/src/dateInput.tsx

```diff
@@ -301,6 +305,8 @@ export class DateInput extends AbstractPureComponent2<DateInputProps, IDateInput
         const { popoverProps = {} } = this.props;
         popoverProps.onClose?.(e);
         this.setState({ isOpen: false });
+        this.startFocusBoundaryElement?.removeEventListener("focusin", this.handleStartFocusBoundaryFocusIn);
```

Comment:
why do we need to use this ref approach and manually bind focusin event handlers?  why can’t it be declarative like normal React code `<div onFocusIn={this.handleEndFocusBoundaryFocusIn} />` ?

Reply:
React doesn't support `onFocusIn`, but it looks like their `onFocus` prop listens to the `focusin` event under the hood. I switched to this approach and everything still works as expected.